### PR TITLE
Renamed Queue to LocalQueue

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?

/kind cleanup
/kind api-change

#### What this PR does / why we need it:

Renames the namespaced resource Queue to LocalQueue.

#### Which issue(s) this PR fixes:
Part of #330

### CODE REVIEW

File: apis/kueue/v1alpha1/localqueue_types.go [hunk 1]
```diff
@@ -42,24 +42,24 @@ type QueueStatus struct {
 //+kubebuilder:printcolumn:name="ClusterQueue",JSONPath=".spec.clusterQueue",type=string,description="Backing ClusterQueue"
 //+kubebuilder:printcolumn:name="Pending Workloads",JSONPath=".status.pendingWorkloads",type=integer,description="Number of pending workloads"
```
Comment (alculquicondor) [inline apis/kueue/v1alpha1/localqueue_types.go:43]: Worth adding a short name:

```
//+kubebuilder:resource:shortName={queue,queues}
```

File: apis/kueue/webhooks/localqueue_webhook.go [hunk 2]
```diff
@@ -30,48 +30,48 @@ import (
...
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(&kueue.Queue{}).
-		WithValidator(&QueueWebhook{}).
+		For(&kueue.LocalQueue{}).
+		WithValidator(&LocalQueueWebhook{}).
 		Complete()
 }
 
-// +kubebuilder:webhook:path=/validate-kueue-x-k8s-io-v1alpha1-queue,mutating=false,failurePolicy=fail,sideEffects=None,groups=kueue.x-k8s.io,resources=queues,verbs=create;update,versions=v1alpha1,name=vqueue.kb.io,admissionReviewVersions=v1
+// +kubebuilder:webhook:path=/validate-kueue-x-k8s-io-v1alpha1-localqueue,mutating=false,failurePolicy=fail,sideEffects=None,groups=kueue.x-k8s.io,resources=localqueues,verbs=create;update,versions=v1alpha1,name=vlocalqueue.kb.io,admissionReviewVersions=v1
```
Comment (alculquicondor) [inline apis/kueue/webhooks/localqueue_webhook.go:44]: :thinking: NB: `kb.io` refers to kubebuilder ¯\_(ツ)_/¯

File: config/manager/kustomization.yaml [hunk 3]
```diff
@@ -14,4 +14,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: gcr.io/k8s-staging-kueue/kueue
-  newTag: main
+  newTag: 0.2.0-devel-159-g59bf53b-dirty
```
Comment (alculquicondor) [inline config/manager/kustomization.yaml:17]: revert (maybe you got a failure when trying to deploy the image)

File: pkg/cache/cache.go [hunk 4]
```diff
@@ -754,12 +754,12 @@ func SetupIndexes(indexer client.FieldIndexer) error {
 	})
 }
 
-func workloadBelongsToQueue(wl *kueue.Workload, q *kueue.Queue) bool {
+func workloadBelongsToQueue(wl *kueue.Workload, q *kueue.LocalQueue) bool {
```
Comment (alculquicondor) [inline pkg/cache/cache.go:757]:
```suggestion
func workloadBelongsToLocalQueue(wl *kueue.Workload, q *kueue.LocalQueue) bool {
```

File: pkg/controller/core/localqueue_controller.go [hunk 5]
```diff
@@ -60,12 +60,12 @@ func (r *QueueReconciler) NotifyWorkloadUpdate(w *kueue.Workload) {
...
 
 //+kubebuilder:rbac:groups="",resources=events,verbs=create;watch;update
-//+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=queues,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=queues/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=queues/finalizers,verbs=update
+//+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=localqueues,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=localqueues/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=localqueues/finalizers,verbs=update
 
 func (r *QueueReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```
Comment (alculquicondor) [inline pkg/controller/core/localqueue_controller.go:67]: LocalQueueReconciler

File: pkg/queue/manager.go [hunk 6]
```diff
@@ -155,16 +155,16 @@ func (m *Manager) DeleteClusterQueue(cq *kueue.ClusterQueue) {
 	m.deleteCohort(cohort, cq.Name)
 }
 
-func (m *Manager) AddQueue(ctx context.Context, q *kueue.Queue) error {
+func (m *Manager) AddQueue(ctx context.Context, q *kueue.LocalQueue) error {
```
Comment (alculquicondor) [inline pkg/queue/manager.go:158]:
```suggestion
func (m *Manager) AddLocalQueue(ctx context.Context, q *kueue.LocalQueue) error {
```

File: pkg/queue/manager.go [hunk 7]
```diff
@@ -180,53 +180,53 @@ func (m *Manager) AddQueue(ctx context.Context, q *kueue.Queue) error {
...
 	cq := m.clusterQueues[qImpl.ClusterQueue]
-	if cq != nil && cq.AddFromQueue(qImpl) {
+	if cq != nil && cq.AddFromLocalQueue(qImpl) {
 		m.Broadcast()
 	}
 	return nil
 }
 
-func (m *Manager) UpdateQueue(q *kueue.Queue) error {
+func (m *Manager) UpdateQueue(q *kueue.LocalQueue) error {
```
Comment (alculquicondor) [inline pkg/queue/manager.go:189]:
```suggestion
func (m *Manager) UpdateLocalQueue(q *kueue.LocalQueue) error {
```

File: pkg/queue/manager.go [hunk 8]
```diff
@@ -180,53 +180,53 @@ func (m *Manager) AddQueue(ctx context.Context, q *kueue.Queue) error {
...
+		if newCQ != nil && newCQ.AddFromLocalQueue(qImpl) {
 			m.Broadcast()
 		}
 	}
 	qImpl.update(q)
 	return nil
 }
 
-func (m *Manager) DeleteQueue(q *kueue.Queue) {
+func (m *Manager) DeleteQueue(q *kueue.LocalQueue) {
```
Comment (alculquicondor) [inline pkg/queue/manager.go:210]:
```suggestion
func (m *Manager) DeleteLocalQueue(q *kueue.LocalQueue) {
```

**Review by kerthcet — COMMENTED** [review]
The localQueue name referenced by other components is still `queueName` or `kueue.x-k8s.io/queue-name`, right?

File: apis/kueue/v1alpha1/localqueue_types.go (hunk 1 above)
Comment (ahg-g) [reply apis/kueue/v1alpha1/localqueue_types.go:43]: should we use that for clusterQueue :)

File: config/manager/kustomization.yaml (hunk 3 above)
Comment (ahg-g) [reply config/manager/kustomization.yaml]: changed by one of the scripts probably

File: apis/kueue/v1alpha1/localqueue_types.go (hunk 1 above)
Comment (alculquicondor) [reply apis/kueue/v1alpha1/localqueue_types.go:43]: Uhm... I don't think so. An end user should be able to do `kubectl get queues` to get the names of the queues they can use in `Workload.spec.queueName`

**Review by alculquicondor — COMMENTED** [review]
nit
/approve

**Review by ahg-g — COMMENTED** [review]
more renames

**Comment by ahg-g** [comment]
> The localQueue name referenced by other components is still `queueName` or `kueue.x-k8s.io/queue-name`, right?

Yes, that will stay as is.

File: apis/kueue/webhooks/localqueue_webhook.go (hunk 2 above)
Comment (ahg-g) [reply apis/kueue/webhooks/localqueue_webhook.go:44]: oh, what is this? should we remove it?

File: pkg/controller/core/localqueue_controller.go [hunk 9]
```diff
@@ -36,17 +36,17 @@ import (
 	"sigs.k8s.io/kueue/pkg/queue"
 )
 
-// QueueReconciler reconciles a Queue object
-type QueueReconciler struct {
+// LocalQueueReconciler reconciles a Queue object
```
Comment (alculquicondor) [inline pkg/controller/core/localqueue_controller.go:39]: I thought I commented on this

```suggestion
// LocalQueueReconciler reconciles a LocalQueue object
```

File: apis/kueue/webhooks/localqueue_webhook.go (hunk 2 above)
Comment (ahg-g) [reply apis/kueue/webhooks/localqueue_webhook.go:44]: there is also `name=mclusterqueue.kb.io` and `name=mresourceflavor.kb.io`

Comment (alculquicondor) [reply apis/kueue/webhooks/localqueue_webhook.go:44]: Let's just keep it, as the code generator would use these patterns if we add more webhooks in the future.

File: pkg/controller/core/localqueue_controller.go (hunk 9 above)
Comment (ahg-g) [reply pkg/controller/core/localqueue_controller.go]: and lots more in this file related to logs

File: config/manager/kustomization.yaml [hunk 10]
```diff
@@ -14,4 +14,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: gcr.io/k8s-staging-kueue/kueue
-  newTag: main
+  newTag: 0.2.0-devel-160-gb92b9c2-dirty
```
Comment (ahg-g) [inline config/manager/kustomization.yaml:17]: this keeps changing when running `make artifacts`

**Comment by ahg-g** [comment]
squashed

**Pull request merged**